### DISCUSSION TITLE
Custom static url for preview deployments

### DISCUSSION THREAD

**Discussion by ystewart** [vercel/community#114] [Help] [answered]
Hi,

I was wondering how to configure a static custom url for preview deployments using self-hosted gitlab.

I want the latest preview (develop branch) to be available on staging.domain.com. Has anyone found a way to do this without using the gitlab integration?

**Comment by Dam-Buty** [comment]
Bumping cause i'm having this exact same issue :pray:

**Reply by JonCognioDigital** [reply]
Also interested in this.

**Reply by marco-streng** [reply]
Same here!

**Reply by philippetedajo** [reply]
same there

**Comment by pcjmfranken** [comment]
Bit late to the party, but the quite well hidden `alias` API still exists... for now:
- Config file option (deprecated): https://vercel.com/docs/configuration?query=alias#project/alias.
- CLI command (strangely undocumented and unlisted): `vercel alias set [vercel deployment url] [desired url]`
- Platform API (not deprecated?): https://vercel.com/docs/api#endpoints/aliases

Chaos! No idea where this API is headed but I wouldn't start using it in production now if you can avoid it.

Unfortunately, the suggested alternative (web UI custom domain settings) is not even close to a replacement with its current limiting featureset.

**Comment by sauntimo** [comment]
I would really like to see this feature implemented. My use case is testing a site which uses firebase; the firebase config requires authorised domains to be specified so I can't log in to my app on preview deployments, rendering them much less helpful than they would otherwise be.

**Reply by go-diego** [reply]
This is my use case exactly

**Comment by JonCognioDigital** [comment]
Ideally, we'd be able to point any custom domain or subdomain to the latest deployment on a chosen branch. Some examples:

www.domain.com => The latest deployment on MASTER branch
www.stagingdomain.com => The latest deployment on the STAGING branch
staging.domain.com => The latest deployment on the STAGING branch
www.devdomain.com => The latest deployment on the DEV branch
dev.domain.com => The latest deployment on the DEV branch

**Reply by fredericoo** [reply]
this is possible, but not preview deployments can be customised as of now — they all have to be on `repo-name-SHA-company.vercel.app`

**Comment by ksyunnnn** [comment]
I I had the same problem.Try this.

https://vercel.com/docs/custom-domains#assigning-a-domain-to-a-git-branch

**Comment by MathiasReker** [comment]
https://vercel.com/docs/api#endpoints/aliases

**Comment by pcjmfranken** [comment, marked as answer]
All tree options from my 2020 post https://github.com/vercel/vercel/discussions/5314#discussioncomment-235223 are still available today, and we've even gained some new ones:

1. Project config file `alias` property - marked legacy (was deprecated, there is a difference), wouldn't start relying on it now:
	https://vercel.com/docs/cli?query=alias#project-configuration/alias

2. CLI Command - has since been properly documented (no longer just our dirty little secret): 
	https://vercel.com/docs/cli?query=alias#commands/alias

3. API Endpoint - maximal freedom, some assembly required:
	https://vercel.com/docs/rest-api#endpoints/aliases

4. Change the default `vercel.app` preview domain - many limitations and costs extra, but it's there if you want it:
	https://vercel.com/docs/concepts/deployments/automatic-urls#preview-deployment-suffix

5. And lastly the custom (sub)domain per Git branch option, which seems to cover all of @ystewart requirements:
	https://vercel.com/docs/concepts/projects/custom-domains#assigning-a-domain-to-a-git-branch

Plenty of methods that are all well documented, even I'm happy. What a time to be alive! (just joking 😝)

**Reply by ghost** [reply]
Thanks! I found this answer helpful. Nice breakdown!

If you're using GitHub actions, you can look at [this example - Vercel: How to alias a preview deployment using the CLI](https://vercel.com/guides/how-to-alias-a-preview-deployment-using-the-cli)
